waspc/src/Wasp/Analyzer/Parser/AbstractParser.hs

```diff
@@ -91,7 +91,7 @@ coerceExpr (SyntaxNode kind width children : remaining)
   | otherwise = do
       startPos <- gets pstateStartPos
       expr <- case kind of
-        S.String -> AST.StringLiteral . tail . init <$> consume width
+        S.String -> AST.StringLiteral . read <$> consume width
```

Comment:
This is the actual fix! Instead of stripping first and last quote, we properly parse it with `read`.

Reply:
@craigmc08 just to check, did I miss something here? Was there maybe a reason why before you used `tail . init` and not `read`? Because now with `read`, it correctly parses string when it has escaped characters in it, while with `tail . init` it didn't.

Reply:
This was a mistake (probably because I didn't write the golden file for the test case correctly). It should be `read`.